File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> checkSet = new HashSet<>();
        Map<Integer, Integer> seenMap = new HashMap<>();
        int ans = 0;

        for(int i : nums){
            checkSet.add(i);
        }

        for(int i : nums){
            if(!seenMap.containsKey(i)){
                int j = i; 
                int curr = 0;
                while(!seenMap.containsKey(j) && checkSet.contains(j)){
                    curr++;
                    checkSet.remove(j);
                    j++;
                }
                if(seenMap.containsKey(j)){
                    curr += seenMap.get(j);
                }
                seenMap.put(i, curr);
                ans = Math.max(ans, curr);
            }
        }

        return ans;
    }
}

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        int n = nums.size();

        vector<int> dp(n, 1);
        int ans = 1;
        for(int i = n-2; i>=0; i--){
            for(int j = i; j<n; j++){
                if(nums[i]<nums[j]){
                    dp[i] = max(dp[i], 1 + dp[j]);
                }
            }
            ans = max(ans, dp[i]);
        }

        return ans;
    }
};
